MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs

```diff
@@ -65,6 +73,10 @@ public bool ValidateVersion(string encoderAppPath, bool logOutput)
                 var yearString = i.ToString(CultureInfo.InvariantCulture);
                 if (output.IndexOf(" " + yearString + " ", StringComparison.OrdinalIgnoreCase) != -1)
                 {
+                    if (logOutput)
+                    {
+                        _logger.LogError("Failed validation checks.  Version year from 2016 or above required: {0}", encoderAppPath);
+                    }
                     return false;
                 }
             }
```

Comment:
I would actually rather we check for real `ffmpeg` version, not some years and stuff... Like we know we support `4.0` only now, and any other version should be checked and reported.

Reply:
We do support 3.2 still as well, for us Debian Stretch users ;-)

Reply:
I don't like the check relying upon the year string and would much prefer to use the version  numbers against the libs, but still searching for docs/release notes detailing what the numbers -> ffmpeg releases are.  Note, I've not modified this behaviour..    Here is the -version' output from ffmpeg:

```
D:\temp>ffmpeg -version
ffmpeg version N-93044-g2e2b44baba Copyright (c) 2000-2019 the FFmpeg developers
built with gcc 8.2.1 (GCC) 20181201
configuration: --enable-gpl <snip>
libavutil      56. 26.100 / 56. 26.100
libavcodec     58. 46.100 / 58. 46.100
libavformat    58. 26.100 / 58. 26.100
libavdevice    58.  6.101 / 58.  6.101
libavfilter     7. 48.100 /  7. 48.100
libswscale      5.  4.100 /  5.  4.100
libswresample   3.  4.100 /  3.  4.100
libpostproc    55.  4.100 / 55.  4.100
```

Reply:
Found it.  Each FFmpeg release details the lib versions contained on here:
https://www.ffmpeg.org/download.html#releases

Reply:
Oh gosh, why `ffmpeg` CLI has to be so... funny?!..

I mean I can see reason here, like one can probably build old enough ffmpeg with new libavcodec and such, but it makes version checking really hard.

@joshuaboniface I'm not saying "don't start if ffmpeg is 3.2", I'm saying "complain in the logs if ffmpeg is not 4.0", as ffmpeg being some other version is known to cause issues (both if it's old and if it's new).

Reply:
Issue #877 raised to cover.

Reply:
@JustAMan Worth stating here that if the FFmpeg version fails validation (currently using year string) then it shall not be used, and the FFmpeg path shall be null and printf to log 'not found'.  This PR has not changed that approach, but it's something we can possibly relax when fixing #877 .